plugins/tiddlywiki/multiwikiserver/modules/commands/mws-add-user.js

```diff
@@ -0,0 +1,58 @@
+/*\
+title: $:/plugins/tiddlywiki/multiwikiserver/commands/mws-add-user.js
+type: application/javascript
+module-type: command
+
+Command to create users and grant permission
+
+\*/
+(function(){
+
+  /*jslint node: true, browser: true */
+  /*global $tw: false */
+  "use strict";
+  
+  var crypto = require("crypto");
+  
+  exports.info = {
+    name: "mws-add-user",
+    synchronous: false
+  };
+  
+  var Command = function(params,commander,callback) {
+    this.params = params;
+    this.commander = commander;
+    this.callback = callback;
+  };
+  
+  Command.prototype.execute = function() {
+    var self = this;
+  
+    if(this.params.length < 2) {
+      return "Usage: --mws-add-user <username> <password> [email]";
+    }
+  
+    if(!$tw.mws || !$tw.mws.store || !$tw.mws.store.sqlTiddlerDatabase) {
+      return "Error: MultiWikiServer or SQL database not initialized.";
+    }
+  
+    var username = this.params[0];
+    var password = this.params[1];
+    var email = this.params[2] || username + "@tiddlywiki.com";
+    var hashedPassword = crypto.createHash("sha256").update(password).digest("hex");
```

Comment:
Simple sha256 hashing the password is not enough. The password needs to be salted and then encrypted at least 10000 times. [Key-stretching](https://en.wikipedia.org/wiki/Key_derivation_function#Key_stretching_and_key_strengthening) makes it harder to use rainbow-tables and brute force password attacks.

Reply:
I'm comfortable with having a placeholder implementation to begin with. At this point, I think clarity is the most important consideration, and with adequate encapsulation it will be straightforward for us to revisit the implementation.

Reply:
The problem is, that it should be done right from the beginning, because the `salt` has to be stored with the user data. It also plays into the login form. So there are at least 3 different positions, where a change here causes changes up to the UI password handling.

Reply:
@pmario the challenge is that we need to merge this PR to get feedback. If we wait until we've got a production ready authentication system then this PR would have to stay open for a long time, making all other work on MWS much more complicated. At this point we have the freedom that if required MWS can nuke the user database on an upgrade. We will lose that freedom as MWS moves beyond the experimental stage, so we should take advantage of it now to improve our overall velocity.

Reply:
OK